### PR TITLE
feat(modal): sets min and max width and height on modal pop-up

### PR DESCRIPTION
**Pull request by gcottrell1**

### CODE REVIEW

File: lib/screen/Modal/Component.tsx [hunk 1]
```diff
@@ -43,21 +57,29 @@ export class ModalScreen extends React.PureComponent<IProps> {
...
-      style.borderWidth = theme.border.width.normal.borderWidth;
-      style.borderRadius = theme.border.radius.large.borderRadius;
+    const dimensionsStyle: ViewStyle = {};
+
+    if (this.state.screenDimensions) {
+      const { height: screenHeight, width: screenWidth } = this.state.screenDimensions;
+      dimensionsStyle.minHeight = screenHeight * 0.5;
+      dimensionsStyle.maxHeight = screenHeight * 0.9;
+      dimensionsStyle.minWidth = screenWidth * 0.5;
+      dimensionsStyle.maxWidth = screenWidth * 0.9;
```
Comment (mattyclarkson) [inline lib/screen/Modal/Component.tsx]: As the element is rendered inside a full screen element can we use percentages in the `styles.ts`?

**Comment by gcottrell1** [comment]
leaving this up as I still think its worth having a minimum width on a modal and removing default styles

File: lib/screen/Modal/Component.tsx [hunk 2]
```diff
@@ -43,21 +57,19 @@ export class ModalScreen extends React.PureComponent<IProps> {
...
-        <View style={[styles.modal, style]}>{node}</View>
+        <View style={[styles.modal, style]} onLayout={this.handleLayoutChange}>{node}</View>
       </View>
     );
   }
+
+  private readonly handleLayoutChange = () => {
+    const { height, width } = Dimensions.get('screen');
+    this.setState({ screenDimensions: { width, height } });
+  }
```
Comment (mattyclarkson) [inline lib/screen/Modal/Component.tsx]: You can just use the percentage values in the `styles`, hopefully you can remove all the dimensions/layout stuff.

File: lib/screen/Modal/styles.ts [hunk 3]
```diff
@@ -15,10 +15,7 @@ export default StyleSheet.create({
     })()),
   },
   modal: {
-    backgroundColor: colour.grey.lightest,
-    borderColor: colour.grey.normal,
-    borderWidth: border.width.normal,
-    borderRadius: border.radius.large,
```
Comment (mattyclarkson) [inline lib/screen/Modal/styles.ts]: I'd rather have a static style override.

**Pull request closed without merging**